src/cpuinfo_x86.c

```diff
@@ -131,6 +129,292 @@ static int IsVendor(const Leaf leaf, const char* const name) {
+    case 0x29:
+      return MakeX86CacheLevelInfo(3, DATA_CACHE, 4 * 1024, 8, 64, -1, 2);
+    case 0x2C:
+      return MakeX86CacheLevelInfo(1, DATA_CACHE, 32, 8, 64, -1, 0);
+    case 0x30:
+      return MakeX86CacheLevelInfo(1, INSTRUCTION_CACHE, 32, 8, 64, -1, 0);
+    case 0x40:
+      return MakeX86CacheLevelInfo(-1, DATA_CACHE, -1, -1, -1, -1, 0);
+    case 0x41:
+      return MakeX86CacheLevelInfo(2, DATA_CACHE, 128, 4, 32, -1, 0);
+    case 0x42:
+      return MakeX86CacheLevelInfo(2, DATA_CACHE, 256, 4, 32, -1, 0);
+    case 0x43:
+      return MakeX86CacheLevelInfo(2, DATA_CACHE, 512, 4, 32, -1, 0);
+    case 0x44:
+      return MakeX86CacheLevelInfo(2, DATA_CACHE, 1 * 1024, 4, 32, -1, 0);
+    case 0x45:
+      return MakeX86CacheLevelInfo(2, DATA_CACHE, 2 * 1024, 4, 32, -1, 0);
+    case 0x46:
+      return MakeX86CacheLevelInfo(3, DATA_CACHE, 4 * 1024, 4, 64, -1, 0);
+    case 0x47:
+      return MakeX86CacheLevelInfo(3, DATA_CACHE, 8 * 1024, 8, 64, -1, 0);
+    case 0x48:
+      return MakeX86CacheLevelInfo(2, DATA_CACHE, 3 * 1024, 12, 64, -1, 0);
+    case 0x49:
+      return MakeX86CacheLevelInfo(2, DATA_CACHE, 4 * 1024, 16, 64, -1, 0);
+    case (0x49 | (1 << 8)):
+      return MakeX86CacheLevelInfo(3, DATA_CACHE, 4 * 1024, 16, 64, -1, 0);
+    case 0x4A:
+      return MakeX86CacheLevelInfo(3, DATA_CACHE, 6 * 1024, 12, 64, -1, 0);
+    case 0x4B:
+      return MakeX86CacheLevelInfo(3, DATA_CACHE, 8 * 1024, 16, 64, -1, 0);
+    case 0x4C:
+      return MakeX86CacheLevelInfo(3, DATA_CACHE, 12 * 1024, 12, 64, -1, 0);
+    case 0x4D:
+      return MakeX86CacheLevelInfo(3, DATA_CACHE, 16 * 1024, 16, 64, -1, 0);
+    case 0x4E:
+      return MakeX86CacheLevelInfo(2, DATA_CACHE, 6 * 1024, 24, 64, -1, 0);
+    case 0x4F:
+      return MakeX86CacheLevelInfo(-1, TLB, 4, -1, -1, 32, 0);
+    case 0x50:
+      return MakeX86CacheLevelInfo(-1, TLB, 4, -1, -1, 64, 0);
+    case 0x51:
+      return MakeX86CacheLevelInfo(-1, TLB, 4, -1, -1, 128, 0);
+    case 0x52:
+      return MakeX86CacheLevelInfo(-1, TLB, 4, -1, -1, 256, 0);
+    case 0x55:
+      return MakeX86CacheLevelInfo(-1, TLB, 2 * 1024, 0xFF, -1, 7, 0);
+    case 0x56:
+      return MakeX86CacheLevelInfo(-1, TLB, 4 * 1024, 4, -1, 16, 0);
+    case 0x57:
+      return MakeX86CacheLevelInfo(-1, TLB, 4, 4, -1, 16, 0);
+    case 0x59:
+      return MakeX86CacheLevelInfo(-1, TLB, 4, 0xFF, -1, 16, 0);
+    case 0x5A:
+      return MakeX86CacheLevelInfo(-1, TLB, 2 * 1024, 4, -1, 32, 0);
+    case 0x5B:
+      return MakeX86CacheLevelInfo(-1, TLB, 4, -1, -1, 64, 0);
+    case 0x5C:
+      return MakeX86CacheLevelInfo(-1, TLB, 4, -1, -1, 128, 0);
+    case 0x5D:
+      return MakeX86CacheLevelInfo(-1, TLB, 4, -1, -1, 256, 0);
+    case 0x60:
+      return MakeX86CacheLevelInfo(1, DATA_CACHE, 16, 8, 64, -1, 0);
+    case 0x61:
+      return MakeX86CacheLevelInfo(-1, TLB, 4, 0xFF, -1, 48, 0);
+    case 0x63:
+      return MakeX86CacheLevelInfo(-1, TLB, 1 * 1024 * 1024, 4, -1, 4, 0);
```

Comment:
I believe this is an error.
You can define constants at the beginning of the function and use it everywhere:
```
const int UNDEF = -1;
const int KiB = 1024;
const int MiB = 1024 * 1024;
...
return MakeX86CacheLevelInfo(UNDEF, TLB, 1 * MiB, 4, UNDEF, 4, 0);
```